src/redis_cmd.cc

```diff
@@ -1602,17 +1602,22 @@ class CommandBPop : public Commander {
 
   rocksdb::Status TryPopFromList() {
     Redis::List list_db(svr_->storage_, conn_->GetNamespace());
-    std::string last_key, elem;
+    std::string elem;
+    std::string* last_key_ptr = nullptr;
     rocksdb::Status s;
     for (const auto &key : keys_) {
-      last_key = key;
+      last_key = &key;
       s = list_db.Pop(key, left_, &elem);
       if (s.ok() || !s.IsNotFound()) {
         break;
       }
     }
     if (s.ok()) {
-      conn_->Reply(Redis::MultiBulkString({last_key, elem}));
+      if (last_key_ptr == nullptr) {
```

Comment:
I think you can avoid comparing with `nullptr` explicitly by replacing
```
if (last_key_ptr == nullptr) {
  conn_->Reply(Redis::MultiBulkString({"", elem}));
} else {
  conn_->Reply(Redis::MultiBulkString({*last_key_ptr, elem}));
}
```
with
```
if (last_key_ptr) {
  conn_->Reply(Redis::MultiBulkString({*last_key_ptr, elem}));
} else {
  conn_->Reply(Redis::MultiBulkString({"", elem}));
}
```

Reply:
I found that sometimes kvrocks uses ` == nullptr`, sometimes just use `(ptr)`, I don't know which is better in this project